satellite/metabase/rangedloop/rangedlooptest/mockprovider.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2022 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package rangedlooptest
+
+import (
+	"context"
+	"math"
+
+	"storj.io/storj/satellite/metabase/rangedloop"
+	"storj.io/storj/satellite/metabase/segmentloop"
+)
+
+// MockProvider allows to iterate over segments from an in-memory source.
```

Comment:
nit: I would call those ProviderMock and RangeProviderMock

Reply:
Changed